src/ebay-menu-button/menu-button.tsx

```diff
@@ -0,0 +1,172 @@
+    onCollapse?: () => void;
+    onExpand?: () => void;
+    onChange?: (i: number, indexes: boolean | boolean[]) => void;
+    onSelect?: () => void;
+    expanded?: boolean;
+    noToggleIcon?: boolean;
+    checked?: number;
+    collapseOnSelect?: boolean;
+    // todo: implement the following props
+    prefixId?: string;
+    prefixLabel?: string;
+}
+
+type Props = Omit<EbayButtonProps, 'type' | 'variant'> &
+    Omit<ComponentProps<'button'>, 'type'> &
+    ComponentProps<'a'> &
+    EbayMenuButtonProps
+
+const EbayMenuButton: FC<Props> = ({
+    type,
+    variant = 'button',
+    className,
+    text = '',
+    fixWidth,
+    reverse,
+    expanded: defaultExpanded = false,
+    noToggleIcon,
+    checked,
+    collapseOnSelect,
+    a11yText,
+    onExpand = () => {},
+    onCollapse = () => {},
+    onChange = () => {},
+    onSelect = () => {},
+    children,
+    ...rest
+}) => {
+    const [expanded, setExpanded] = useState(defaultExpanded)
+    const [menuId, setMenuId] = useState<string | undefined>()
+    const buttonRef = useRef(null)
+    const menuRef = useRef()
+
+    const menuItems = filterByType(children, [EbayMenuButtonItem, EbayMenuButtonSeparator])
+    const [checkedIndexes, setCheckedIndexes] = useState<boolean[]>(menuItems.map(() => false))
+
+    const label = findComponent(children, EbayMenuButtonLabel) || <span>{text}</span> || null
+    const icon = findComponent(children, EbayMenuButtonIcon)
+    const wrapperClasses = classnames('menu-button', className)
+    const menuClasses = classnames('menu-button__menu', {
+        'menu-button__menu--fix-width': fixWidth,
+        'menu-button__menu--reverse': reverse
+    })
+
+    useEffect(() => {
+        const handleBackgroundClick = (e: React.MouseEvent) => {
+            const menuEl = menuRef.current as HTMLDivElement
+            const menuClicked = menuEl && menuEl.contains(e.target as Node)
+            if (collapseOnSelect || !menuClicked) {
+                setExpanded(false)
+            }
+        }
+
+        if (expanded) {
+            onExpand()
+            // On React 18 useEffect hooks runs synchronous instead of asynchronous as React 17 or prior
+            // causing the event listener to be attached to the document at the same time that the dialog
+            // opens. Adding a timeout so the event is attached after the click event that opened the modal
+            // is finished.
+            setTimeout(() => {
+                document.addEventListener('click', handleBackgroundClick as any, false)
+            })
+        } else if (expanded === false) {
+            onCollapse()
+        }
+        return () => document.removeEventListener('click', handleBackgroundClick as any, false)
+    }, [expanded])
+
+    useEffect(() => {
+        setMenuId(randomId())
+    }, [])
+
+    const handleMenuKeydown = (_, __, e) => {
+        handleEscapeKeydown(e, () => {
+            setExpanded(false)
+            buttonRef.current?.focus()
```

Comment:
Strangely, the focus does not return to the button for variant overflow.

- See storybook: https://opensource.ebay.com/ebayui-core-react/EbayMenuButton-MVP/index.html?path=/story/ebay-menu-button--variant-overflow
- Use keyboard to open the menu
- Press Esc to close the menu
- Focus does not return to the icon button (focus should be on the icon button)